JoinColony/colonyCDapp
# Feat: enable Payment Stream cancellation via a Motion

## Description

Here are the required block-ingestor updates: [Feat: enable handler for the cancel streaming payment motion](https://github.com/JoinColony/block-ingestor/pull/240)

## Testing

1. Install the "Reputation Weighted" extension and use the "Testing governance" preset
2. Install the "Streaming Payments" extension
3. Using the debugging interface, create a Streaming Payment
![Screenshot 2024-06-14 at 10 56 11](https://github.com/JoinColony/colonyCDapp/assets/50642296/ca2ec05f-63f1-4779-9119-bafed8c315c6)
4. Wait for 3-5 seconds, then click the "Use latest Streaming Payment Native ID" button
![Screenshot 2024-06-14 at 10 57 42](https://github.com/JoinColony/colonyCDapp/assets/50642296/1af0f69c-5470-46ad-b20e-8a5a9369f657)
5. Copy the new value in the Streaming Payments input field, this should be the Native ID of the Streaming Payment you just made
6. Open the GraphQL playground http://localhost:20002/ and run the following query:
> Replace NATIVE_ID with the Native ID you just copied.
> This query should show you the Streaming Payment you just made.
```gql
query ListStreamingPayments {
  listStreamingPayments(filter: {nativeId: {eq: NATIVE_ID}}) {
    items {
      isCancelled
    }
  }
}
```
7. Verify that the `isCancelled` field is set to `null`
![Screenshot 2024-06-17 at 07 37 55](https://github.com/JoinColony/colonyCDapp/assets/50642296/09e70e61-8376-493b-88a3-4450d9ab77ef)
8. Click the "Cancel via motion" button
![Screenshot 2024-06-14 at 11 01 52](https://github.com/JoinColony/colonyCDapp/assets/50642296/741ae79d-6882-4c8c-8879-42846110cc6a)
9. Wait for 3-5 seconds then click this button to open the action you just made, then copy the Transaction Hash from the URL
![Screenshot 2024-06-17 at 07 31 47](https://github.com/JoinColony/colonyCDapp/assets/50642296/da9411e0-2efa-49db-8739-e2a8d71335ec)
10. Run the following query:
> Replace TRANSACTION_HASH with the Transaction Hash you just copied.
> This should show you both the Streaming Payment you made and the motion you made
```gql
query ListStreamingPayments {
  listStreamingPayments(filter: {nativeId: {eq: NATIVE_ID}}) {
    items {
      isCancelled
    }
  }
  listColonyMotions(
    filter: {transactionHash: {eq: TRANSACTION_HASH}}
  ) {
    items {
      isFinalized
    }
  }
}
```
11. Make sure that:
- The Streaming Payment's `isCancelled` field is set to `null`
- The motion's `isFinalized` field is set to `false`
![Screenshot 2024-06-14 at 10 53 02](https://github.com/JoinColony/colonyCDapp/assets/50642296/e2cef75b-1d8f-41ef-83bb-4a967e890b97)
12. Open the User Hub transaction entry for the "Cancel Streaming Payment" motion you started
13. Approve the motion and back it up with the maximum stake
14. Advance the time by running the following commands:
```bash
curl -X POST --data '{"jsonrpc":"2.0","method":"evm_increaseTime","params":[3600],"id":1}' localhost:8545
curl -X POST --data '{"jsonrpc":"2.0","method":"evm_mine","params":[],"id":1}' localhost:8545
```
15. Reload the page
16. Finalise the motion
17. Run the query on step 10 to get an update for your Streaming Payment and the motion you made
18. Verify that:
- The Streaming Payment's `isCancelled` field is set to `true`
- The motion's `isFinalized` field is set to `true`
![Screenshot 2024-06-14 at 10 53 55](https://github.com/JoinColony/colonyCDapp/assets/50642296/0e5379c9-d467-4be2-876d-d3a6b29e937d)

## Diffs

**New stuff** ✨

* A new saga for handling the "Cancel Streaming Payment" motion

Resolves #2216
